### PR TITLE
Fix align_to_coord and plot

### PR DESCRIPTION
**Pull request by d-chambers**
## Description
This PR changes `align_to_coord`  (see #561 and #578) so that it has more validation checks and only works in a "relative" mode, which makes more sense as the mixing usage of relative and absolute didn't quite fit. This is a breaking change, but since `align_to_coord` is only in master it doesn't matter. 

This also modifies the new waterfall plot (#574)  such that `timedelta64` y axis are not formatted to look like datetime64 axis. 

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [ ] documented the new feature with docstrings and/or appropriate doc page.
- [ ] included tests. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation and clearer errors for alignment coordinates and sample handling; more consistent time-axis formatting.

* **Refactor**
  * Alignment logic simplified to operate uniformly on coordinate values, removing legacy relative/absolute branching and reducing edge-case complexity.

* **New Features**
  * Waterfall visualization: added interpolation option and improved image resampling behavior.

* **Tests**
  * Alignment tests reorganized: added new validations and round‑trip checks; removed/renamed outdated cases to match new coordinate semantics.

* **Documentation**
  * Clarified docstrings and examples for alignment and mute APIs.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: dascore/proc/align.py [hunk 1]
```diff
@@ -174,61 +176,86 @@ def _validate_alignment_inputs(patch, kwargs):
...
+        relative=False if samples else True,
     )
-    out = inds - np.min(inds)
-    # Reverse index. This way the min value still is at 0 (reference)
-    if reverse:
-        out *= -1
-    return out
+    # Reverse index if needed. This way the reference stays the same.
+    inds = inds_abs * (sign_vals.astype(np.int64) * (-1 if reverse else 1))
+    return inds
```
[the remark this answers is not in the document]
Comment (d-chambers) [reply dascore/proc/align.py]: No, this is ok. You are right that `np.sign(timedelta64)` returns a timedelta64 array. However, its values are {1, -1} so the astype will properly convert to an int array of {1, -1} since the values can't be anything different in the timedelta64 array.
